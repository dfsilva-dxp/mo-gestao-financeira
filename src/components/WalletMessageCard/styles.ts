import styled, { css } from "styled-components";

export const Wrapper = styled.div`
  ${({ theme }) => css`
    background: ${theme.colors.gradient};
    width: 34rem;
    padding: ${theme.spacings.small} ${theme.spacings.large};
    border-radius: ${theme.border.radius.default};
  `}
`;

export const Paragraph = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};

    display: grid;
    grid-template-rows: repeat(2, auto);
    place-items: center;
  `}
`;

export const Span = styled.span`
  ${({ theme }) => css`
    font-size: 1.3rem;
    color: ${theme.colors.gray};
  `}
`;
