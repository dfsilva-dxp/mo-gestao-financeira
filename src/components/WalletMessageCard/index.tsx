import { Paragraph, Span, Wrapper } from "./styles";

export default function WalletMessageCard() {
  return (
    <Wrapper>
      <Paragraph>
        Muito bem, sua carteira est√° positiva!
        <Span>Considere investir o seu saldo!</Span>
      </Paragraph>
    </Wrapper>
  );
}
