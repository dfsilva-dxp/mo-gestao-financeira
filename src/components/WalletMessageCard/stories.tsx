import { Story, Meta } from "@storybook/react/types-6-0";
import { ComponentProps } from "react";

import WalletMessageCard from ".";

export default {
  title: "WalletMessageCard",
  component: WalletMessageCard,
  argTypes: {},
  parameters: {
    layout: "centered"
  },
  decorators: [
    (Story) => (
      <div style={{ margin: "0 auto" }}>
        <Story />
      </div>
    )
  ]
} as Meta;

export const Basic: Story<ComponentProps<typeof WalletMessageCard>> = () => (
  <WalletMessageCard />
);
