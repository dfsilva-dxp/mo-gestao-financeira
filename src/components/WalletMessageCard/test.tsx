import { screen } from "@testing-library/react";
import { renderWithTheme } from "utils/tests/helpers";

import WalletMessageCard from ".";

describe("<WalletMessageCard />", () => {
  it("should render currectly", () => {
    const { container } = renderWithTheme(<WalletMessageCard />);

    expect(
      screen.getByText(/Muito bem, sua carteira está positiva!/i).parentElement
    ).toHaveStyle({
      background:
        "linear-gradient(145deg, rgba(33,34,45,1) 0%, rgba(58,44,58,1) 100%)",
      width: "34rem"
    });

    expect(
      screen.getByText(/Muito bem, sua carteira está positiva!/i)
    ).toBeInTheDocument();

    expect(
      screen.getByText(/Muito bem, sua carteira está positiva!/i)
    ).toHaveStyle({
      "font-size": "1.4rem"
    });

    expect(
      screen.getByText(/Considere investir o seu saldo!/i)
    ).toBeInTheDocument();

    expect(screen.getByText(/Considere investir o seu saldo!/i)).toHaveStyle({
      color: "#797D9A",
      "font-size": "1.3rem"
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
