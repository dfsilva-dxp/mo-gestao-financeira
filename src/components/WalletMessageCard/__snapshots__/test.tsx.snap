// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WalletMessageCard /> should render currectly 1`] = `
.c0 {
  background: linear-gradient(145deg,rgba(33,34,45,1) 0%,rgba(58,44,58,1) 100%);
  width: 34rem;
  padding: 2.0rem 4.0rem;
  border-radius: 1.6rem;
}

.c1 {
  font-size: 1.4rem;
  display: grid;
  grid-template-rows: repeat(2,auto);
  place-items: center;
}

.c2 {
  font-size: 1.3rem;
  color: #797D9A;
}

<div
  class="c0"
>
  <p
    class="c1"
  >
    Muito bem, sua carteira est√° positiva!
    <span
      class="c2"
    >
      Considere investir o seu saldo!
    </span>
  </p>
</div>
`;
